examples/rustls-mtls/Cargo.toml

```
@@ -0,0 +1,17 @@
+[package]
+name = "rustls-mtls"
+version = "0.1.0"
+edition = "2021"
+authors = ["Rick Richardson <rick.richardson@gmail.com>"]
+
+[dependencies]
+# Remove the `provider-tls-default` feature and add `provider-tls-rustls`
+s2n-quic = { version = "1", path = "../../quic/s2n-quic", default-features = false, features = ["provider-address-token-default", "provider-tls-rustls"] }
```

dont think you need this:

```suggestion
s2n-quic = { version = "1", path = "../../quic/s2n-quic", default-features = false, features = ["provider-tls-rustls"] }
```

---

It appears that a bunch of code in s2n-quic that fails to compile if `provider-address-token-default` is not included.  
A handful of dependencies don't get included. 

That should definitely get fixed, but I think it's outside the scope of this PR.

---

Specifically:   `s2n-quic/src/provider/address_token/default.rs` is included regardless of whether Feature `provider-address-token-default` is set.  However, that feature gates a few dependencies that are necessary for `address_token/default.rs` to compile.

---

`provider-address-token-default` is a default feature,   maybe that just shouldn't be a feature at all?